src/foursquare-lunch.coffee

```diff
@@ -22,22 +28,75 @@ module.exports = (robot) ->
     redirectUrl: "localhost"
   config.version = '20150722'
 
-  foursquare = require('node-foursquare')(config);
+  locates = () -> robot.brain.data.lunch_locates ?= {}
+
+  # locates()[key]
+  # locates()[key]
+  # locates()[key] = value
+
+  foursquare = require('node-foursquare')(config)
 
-  robot.respond /lunch$/i, (msg) ->
-    suggestLunchSpot msg
+  robot.respond /lunch/i, (msg) ->
+    from_user = msg.message.user.name
+
+    if _match = msg.message.text.match('([0-9\.]+) mile')
+      meters = parseInt(_match[1] * 1609.344)
+    if _match = msg.message.text.match('([0-9\.]+) meter')
+      meters = parseInt _match[1]
+    if _match = msg.message.text.match('near (.*)')
```

Comment:
Dropping a note here that the new commands need to be added to the `README.md` file.